File: src/main/java/com/example/ProyectoFinal/Repository/AutorRepository.java
package com.example.ProyectoFinal.Repository;
import com.example.ProyectoFinal.Model.Autor;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface AutorRepository extends JpaRepository<Autor, Long> {}

File: app/src/main/java/com/example/proyectofinal/Repository/AutorRepository.kt
package com.example.proyectofinal.Repository

import com.example.proyectofinal.interfaces.AutorRetrofitClient
import com.example.proyectofinal.model.Autor

class AutorRepository {

    suspend fun getAutores(): List<Autor> {
        return AutorRetrofitClient.autorApi.getAutores()
    }

    suspend fun getAutorById(id: Long): Autor {
        return AutorRetrofitClient.autorApi.getAutorById(id)
    }

    suspend fun crearAutor(autor: Autor): Autor {
        return AutorRetrofitClient.autorApi.crearAutor(autor)
    }

    suspend fun eliminarAutor(id: Long) {
        AutorRetrofitClient.autorApi.eliminarAutor(id)
    }
}
